File: PHP/bot.php
require_once __DIR__ . '/vendor/autoload.php';

use Discord\Discord;
use Discord\Parts\Channel\Message;

$discord = new Discord([
    'token' => 'YOUR_DISCORD_BOT_TOKEN_HERE',
]);

$discord->on('ready', function ($discord) {
    echo "Bot is ready!", PHP_EOL;
});

$discord->on('message', function (Message $message, Discord $discord) {
    if ($message->content == '!hello') {
        $message->reply('Hello, world!');
    }
});

$discord->run();

File: Rust/bot.rs
use serenity::{
    async_trait,
    model::{channel::Message, gateway::Ready},
    prelude::*,
};

struct Handler;

#[async_trait]
impl EventHandler for Handler {
    async fn message(&self, ctx: Context, msg: Message) {
        if msg.content == "!hello" {
            if let Err(why) = msg.channel_id.say(&ctx.http, "hello world").await {
                println!("Error sending message: {:?}", why);
            }
        }
    }

    async fn ready(&self, _: Context, _: Ready) {
        println!("Bot is ready!");
    }
}

#[tokio::main]
async fn main() {
    let token = "your_token_here";

    let mut client = Client::builder(token)
        .event_handler(Handler)
        .await
        .expect("Error creating client");

    if let Err(why) = client.start().await {
        println!("Client error: {:?}", why);
    }
}
